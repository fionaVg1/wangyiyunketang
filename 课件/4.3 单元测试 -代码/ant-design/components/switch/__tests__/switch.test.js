import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Switch from '..';
import { wrap } from 'lodash';

describe('Switch', () => {

  it('should match snapshot', () => {
    const wrapper = shallow(<Switch />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('should checked when padd checked=true props', () => {
    const wrapper = render(<Switch checked={true}/>);
    expect(wrapper.hasClass('ant-switch-checked')).toBe(true);
  });

  it('should render checked children correctly', () => {
    const wrapper = mount(<Switch checked={true} checkedChildren={'ok'}/>);
    expect(wrapper.text()).toBe('ok');
  });

  it('should be disable when pass props disable=true', () => {
    const wrapper = render(<Switch disabled={true}/>);
    expect(wrapper.hasClass('ant-switch-disabled')).toBe(true);
  });

  it('should call onChange function when switch toggled', () => {
    const change = jest.fn(checked => checked);
    const wrapper = mount(<Switch onChange={change} />);
    wrapper.simulate('click');
    expect(change.mock.results[0].value).toBe(true);
    wrapper.simulate('click');
    expect(change.mock.results[1].value).toBe(false);
  })

});
