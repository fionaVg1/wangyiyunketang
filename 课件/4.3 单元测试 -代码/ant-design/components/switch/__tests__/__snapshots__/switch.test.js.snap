// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should match snapshot 1`] = `
<button
  aria-checked="false"
  class="ant-switch"
  role="switch"
  type="button"
>
  <div
    class="ant-switch-handle"
  />
  <span
    class="ant-switch-inner"
  />
</button>
`;
